header.go

```diff
@@ -60,6 +60,9 @@ func UncompressAllocHdr(out, in []byte) ([]byte, error) {
 		return out, errTooShort
 	}
 	origlen := binary.LittleEndian.Uint32(in)
+	if origlen <= 0 {
+		return out, nil
+	}
```

Comment:
Have you tested this? Looking at the control flow of the function, this will never trigger: the function would return `errTooShort`.

Reply:
This is to test if there is a header whose value is 0. So there are 4 bytes present for the header, but its uint32 value is 0.

Reply:
Oh, I see!. It seems to me like this should return `errTooShort` or another error then, wdyt?

Reply:
I _think_ this is a valid case where this is a payload for which there is null content, so I don't think an error is appropriate. Just returning the null content back makes more sense to me

Reply:
For clarity, without this change, `UncompressAllocHdr` returns an error: `Decompression failed: Malformed compression stream` I think this is arguably the correct result? We wrap `LZ4_decompress_safe` with these functions, and the error coms from it. Its documentation says:

```
compressedSize : is the exact complete size of the compressed block.
...
If the source stream is detected malformed, the function will stop decoding and return a negative result.
This function is protected against malicious data packets.
```

https://github.com/lz4/lz4/blob/v1.8.3/lib/lz4.h#L141-L148

So this error happens because a caller passed a buffer that is too long to `UncompressAllocHdr`.

Additionally: LZ4 produces output for compressing this message. The buffer contents are: 5 `0x00` bytes. Passing anything other than those 5 bytes should be an error?

The same problem happens with other inputs. E.g. if I pass in `"abc"`, LZ4 outputs 8 bytes. If I pass longer buffers, we get an error. I think that is correct?

Reply:
Fair points to be sure. For context, this came up when decompressing lz4 payloads from dogweb, which seems like behaves a bit differently than this impl when it comes to empty inputs. And yeah, Malformed compression stream is the error that we get.

Reply:
Also, I'm not actually sure what makes sense in this empty payload case anyway. Compressing a zero-length message should arguably be 4 `0x00` bytes instead of 5, right? But AFAIK we're not following any sort of actual standard here with the header format--just a de-facto standard.

Reply:
You can see the root of the problem in this PR which fixes it: https://github.com/DataDog/python-lz4/pull/1/files

TLDR is the python-lz4 lib leaves garbage bytes at the end of its output when given a zero-length input.

Now I think there's a debate whether it makes sense to fix this with the python-lz4 fork, which would be a new fork, or to just fix it here.

More context and discussion here: https://github.com/DataDog/dogweb/pull/73067